### DISCUSSION TITLE
FIRE.ADD_OWNED_EXPLOSION triggers anticheat

### DISCUSSION THREAD

**Discussion by pierrelasse** [YimMenu/YimMenu#2210] [Q&A] [answered]
So i stumbled upon [FIRE.ADD_OWNED_EXPLOSION](https://docs.fivem.net/natives/?_0x172AA1B624FA1013) and when i tried to use it, i got kicked for some network error. Then i searched for it in yimmenu and found the following:
```c++
inline void blame_explode_coord(player_ptr to_blame, Vector3 pos, eExplosionTag explosion_type, float damage, bool is_audible, bool is_invisible, float camera_shake)
	{
		explosion_anti_cheat_bypass::m_can_blame_others->apply();
		explosion_anti_cheat_bypass::m_can_use_blocked_explosions->apply();

		FIRE::ADD_OWNED_EXPLOSION(
		    (*g_pointers->m_gta.m_is_session_started && to_blame) ? PLAYER::GET_PLAYER_PED_SCRIPT_INDEX(to_blame->id()) : 0,
		    pos.x,
		    pos.y,
		    pos.z,
		    (int)explosion_type,
		    damage,
		    is_audible,
		    is_invisible,
		    camera_shake);

		explosion_anti_cheat_bypass::m_can_use_blocked_explosions->restore();
		explosion_anti_cheat_bypass::m_can_blame_others->restore();
	}
```

Now i'm wondering if the anticheat is kicking me or smth? And if yes how to bypass this in a lua script.

**Comment by maybegreat48** [comment, marked as answer]
Yes. Using that native without a bypass will kick you out of the session and send a report to Rockstar

**Reply by pierrelasse** [reply]
://////oof.
Since yimmenu explicitly defines that function, yim could also define a function that is already bypassing right?

**Comment by pierrelasse** [comment]
I saw that kiddions has a own version of fire.add_explosion.
Maybe this could also be added to yimmenu.

**Reply by maybegreat48** [reply]
> I saw that kiddions has a own version of fire.add_explosion.

I doubt it

**Reply by pierrelasse** [reply]
> I doubt it

First: if or if not, yimmenu could still add it
Second:
https://www.youtube.com/watch?v=j4ovbmsp6p0
They use it idk
https://raw.githubusercontent.com/Keramis/2t1-Sneaky-Explodes/main/Sneaky%20Explodes.V.3.lua

**Comment by pierrelasse** [comment]
Closed due to https://github.com/YimMenu/YimMenu/issues/2254
